### DISCUSSION TITLE
Cannot call method toArray() on string when using AsEnumArrayObject

### DISCUSSION THREAD

**Discussion by joostdebruijn** [larastan/larastan#2103] [Q&A] [answered]
Hi all,

I've the following cast on my model:

```php

use App\Enums\ServerStatus;
use Illuminate\Database\Eloquent\Casts\AsEnumArrayObject;

 /**
  * @return array{status: 'Illuminate\Database\Eloquent\Casts\ArrayObject<array-key, \App\Enums\ServerStatus>'}
  */
protected function casts(): array
{
    return [
        'status' => AsEnumArrayObject::of(ServerStatus::class),
    ];
}
```

When calling for example `$this->status->toArray()` I receive: `Cannot call method toArray() on string.`. Probably I'm using a wrong annotation, but I cannot find the a correct way to do this. Can someone help me out?

**Comment by szepeviktor** [comment]
https://laravel.com/docs/11.x/eloquent-mutators#casting-arrays-of-enums

**Reply by joostdebruijn** [reply]
Yep, you're right Viktor. I was editing my original code. The example is correct now. Sorry for the inconvenience.

**Comment by szepeviktor** [comment]
maybe one backslash more
```php
 /**
  * @return array{status: '\Illuminate\Database\Eloquent\Casts\ArrayObject<array-key, \App\Enums\ServerStatus>'}
  */
```

**Reply by joostdebruijn** [reply]
Thanks for the quick responses. I tried both (with and without backslash) and the error persists. I've cleared the PHPstan cache in both runs.

**Comment by szepeviktor** [comment]
Our tests may help you.
https://github.com/larastan/larastan/blob/df279123c61996dcd8feabe4a3534184d82fc1eb/tests/application/app/User.php#L56

**Reply by joostdebruijn** [reply, marked as answer]
I switched to the following notation:

```php
protected $casts = [
  'status' => AsEnumArrayObject::class.':'.ServerStatus::class,
];
```
That works properly. 😄

**Reply by szepeviktor** [reply]
You get many rockets 🚀🚀🚀
